client/src/components/turnProvider/turnProvider.tsx

```diff
@@ -0,0 +1,35 @@
+import { createContext, useState, useContext } from "react";
+import { ChildrenProps } from "@/constants/game";
+import { Dispatch, SetStateAction } from "react";
+import { error } from "console";
+export const isMyTurn = (lastPlayer:string | null | undefined) => {
+    return lastPlayer !== sessionStorage.getItem('player')
+}
+
+interface TurnContextProps {
+    turn: boolean,
+    setTurn:  Dispatch<SetStateAction<boolean>>
```

Comment:
What is Dispatch doing here? Could please help me understand its use with some doc/article?

Reply:
this is type of the setState function returned by the useState hook in react.

Reply:
useState: https://react.dev/reference/react/useState